src/components/ConfigItem/ConfigItem.stories.tsx

```diff
@@ -0,0 +1,72 @@
+import { useState } from "react";
+import { ConfigItem } from "./index";
+import { ConfigDB } from "../../api/axios";
+import { MOCK_DATA } from "./storybook-mock";
+
+ConfigDB.get = (url: string) => {
+  if (url.indexOf("Subnet") > -1) {
+    return Promise.resolve({
+      data: MOCK_DATA.filter((v) => v.config_type === "Subnet")
+    });
+  } else if (url.indexOf("EC2Instance") > -1) {
+    return Promise.resolve({
+      data: MOCK_DATA.filter((v) => v.config_type === "EC2Instance")
+    });
+  } else if (url.indexOf("/config_item?id=eq.") > -1) {
+    const id = url.replace("/config_item?id=eq.", "");
+    return Promise.resolve({
+      data: [MOCK_DATA.find((item) => item.id === id)]
+    });
+  }
+};
+
+const ConfigItemDropDown = ({ type }: { type: string }) => {
+  const [selectedItem, setSelectedItem] = useState(null);
+  const [dependentSelectedItem, setDependentSelectedItem] = useState(null);
+  return (
+    <div className="flex flex-col">
+      <ConfigItem
+        className="w-96"
+        type={type}
+        value={selectedItem}
+        autoFetch={true}
+        onSelect={(item) => {
+          setSelectedItem(item);
+          setDependentSelectedItem(null);
+        }}
+        dependentConfigItems={[
```

Comment:
Shouldn't this be a child rather than a property?

Reply:
We can do that, I just went with this approach to do solve it first. Adopting this to children way of loading is straightforward. I am working on it.